scipy/stats/tests/test_resampling.py

```diff
@@ -732,124 +737,197 @@ def statistic_extradim(*args, axis):
     @pytest.mark.parametrize('axis', range(-3, 3))
-    def test_axis(self, axis):
+    def test_axis_dtype(self, axis, xp):
         # test that Nd-array samples are handled correctly for valid values
-        # of the `axis` parameter
+        # of the `axis` parameter; also make sure non-default dtype is maintained
         rng = np.random.default_rng(2389234)
-        norm_rvs = self.rvs(stats.norm.rvs, rng)
-
         size = [2, 3, 4]
         size[axis] = 100
-        x = norm_rvs(size=size)
-        expected = stats.skewtest(x, axis=axis)
 
-        def statistic(x, axis):
-            return stats.skewtest(x, axis=axis).statistic
-
-        res = monte_carlo_test(x, norm_rvs, statistic, vectorized=True,
+        # Determine non-default dtype
+        dtype_default = xp.asarray(1.).dtype
+        dtype_str = 'float32'if ("64" in str(dtype_default)) else 'float64'
+        dtype_np = getattr(np, dtype_str)
+        dtype = getattr(xp, dtype_str)
+
+        # ttest_1samp is CPU array-API compatible, but it would be good to
+        # include CuPy in this test. We'll perform ttest_1samp with a
+        # NumPy array, but all the rest with be done with fully array-API
+        # compatible code.
+        x = rng.standard_normal(size=size, dtype=dtype_np)
+        expected = stats.ttest_1samp(x, popmean=0., axis=axis)
+
+        x = xp.asarray(x, dtype=dtype)
+        xp_test = array_namespace(x)  # numpy.std doesn't have `correction`
+        statistic = self.get_statistic(xp_test)
+        rvs = self.get_rvs(stats.norm.rvs, rng, dtype=dtype, xp=xp)
+
+        res = monte_carlo_test(x, rvs, statistic, vectorized=True,
                                n_resamples=20000, axis=axis)
 
-        assert_allclose(res.statistic, expected.statistic)
-        assert_allclose(res.pvalue, expected.pvalue, atol=self.atol)
+        ref_statistic = xp.asarray(expected.statistic, dtype=dtype)
+        ref_pvalue = xp.asarray(expected.pvalue, dtype=dtype)
+        xp_assert_close(res.statistic, ref_statistic)
+        xp_assert_close(res.pvalue, ref_pvalue, atol=self.atol)
+
+    @array_api_compatible
+    @pytest.mark.parametrize('alternative', ("two-sided", "less", "greater"))
+    def test_alternative(self, alternative, xp):
+        # test that `alternative` is working as expected
+        rng = np.random.default_rng(65723433)
+
+        x = rng.standard_normal(size=30)
+        ref = stats.ttest_1samp(x, 0., alternative=alternative)
+
+        x = xp.asarray(x)
+        xp_test = array_namespace(x)  # numpy.std doesn't have `correction`
+        statistic = self.get_statistic(xp_test)
+        rvs = self.get_rvs(stats.norm.rvs, rng, xp=xp)
+
+        res = monte_carlo_test(x, rvs, statistic, alternative=alternative)
+
+        xp_assert_close(res.statistic, xp.asarray(ref.statistic))
+        xp_assert_close(res.pvalue, xp.asarray(ref.pvalue), atol=self.atol)
+
 
     @pytest.mark.slow
     @pytest.mark.parametrize('alternative', ("less", "greater"))
```

Comment:
Can you add a comment explaining why the tests below haven't been converted? I guess they are waiting on other things?

Reply:
Sure; suggested above.